src/topology/mod.rs

```diff
@@ -164,33 +176,9 @@ impl RunningTopology {
             }
         };
 
-        fn to_remove_change_add<C>(
-            old: &IndexMap<String, C>,
-            new: &IndexMap<String, C>,
-        ) -> (HashSet<String>, HashSet<String>, HashSet<String>)
-        where
-            C: serde::Serialize + serde::Deserialize<'static>,
-        {
-            let old_names = old.keys().cloned().collect::<HashSet<_>>();
-            let new_names = new.keys().cloned().collect::<HashSet<_>>();
-
-            let to_change = old_names
-                .intersection(&new_names)
-                .filter(|&n| {
-                    // This is a hack around the issue of comparing two
-                    // trait objects. Json is used here over toml since
-                    // toml does not support serializing `None`.
-                    let old_json = serde_json::to_vec(&old[n]).unwrap();
-                    let new_json = serde_json::to_vec(&new[n]).unwrap();
-                    old_json != new_json
-                })
-                .cloned()
-                .collect::<HashSet<_>>();
-
-            let to_remove = &old_names - &new_names;
-            let to_add = &new_names - &old_names;
-
-            (to_remove, to_change, to_add)
+        if dry_run && !require_healthy {
```

Comment:
Do we need to worry about `dry_run` in the case where we're reloading? Since it won't change over the lifetime of the process and we won't start the topology if it's true, it doesn't seem like we'd ever need to worry about it being true when we're in a reload situation.

Reply:
Yeah, but it's actually generic (and private) method for both cases, it's named `load_config_and_respawn` now.
Higher-level `reload_config_and_respawn` made public and doesn't contain `dry_run` and `initial_load` arguments.